tests_and_analysis/test/euphonic_test/test_force_constants_calculate_qpt_phonons_phonopy.py

```diff
@@ -0,0 +1,138 @@
+             12.20855918, 12.20855918, 12.23105123, 12.88310822, 12.88310822,
+             12.89476859, 12.89476859, 13.12198284, 13.13553123, 13.13553123,
+             13.13693891, 22.04350987, 22.05424945, 22.05424945, 22.06651219,
+             29.19174209, 29.19174209, 29.19468408, 29.1978153],
+            [8.65321026, 8.65334846, 9.98387815, 9.98421216, 12.84875298,
+             12.85137816, 12.86815343, 12.90260046, 13.51181836, 13.51450714,
+             13.69142224, 13.69831554, 13.76153287, 13.76224884, 14.51838141,
+             14.5334654, 19.37183479, 19.39110497, 21.03444816, 21.04652258,
+             28.28732511, 28.28804436, 29.77604856, 29.77629323]
+        ]) * ureg('meV')
+    }
+
+    @pytest.mark.parametrize(
+        ("kwargs"), [
+            {"dipole": True, "splitting": False},
+            {
+                "dipole": True, "splitting": False, "asr": None,
+                "use_c": True, "fall_back_on_python": False
+            },
+            {
+                "dipole": True, "splitting": False,
+                "use_c": True, "fall_back_on_python": False, "n_threads": 2
+            },
+            {"dipole": True, "splitting": False, "asr": 'reciprocal'},
+            {
+                "dipole": True, "splitting": False, "asr": 'reciprocal',
+                "use_c": True, "fall_back_on_python": False
+            },
+            {
+                "dipole": True, "splitting": False, "asr": 'reciprocal',
+                "use_c": True, "fall_back_on_python": False, "n_threads": 2
+            },
+            {"dipole": True, "splitting": False, "asr": 'realspace'},
+            {
+                "dipole": True, "splitting": False, "asr": 'realspace',
+                "use_c": True, "fall_back_on_python": False
+            },
+            {
+                "dipole": True, "splitting": False, "asr": 'realspace',
+                "use_c": True, "fall_back_on_python": False, "n_threads": 2
+            }
+        ]
+    )
+    def test_phonopy_calculate_qpoint_phonon_modes(self, kwargs):
+        fc = ForceConstants.from_phonopy(
+            path=self.phonopy_yaml_file, summary_name="phonopy.yaml"
+        )
+        if "asr" in kwargs and kwargs["asr"] == 'realspace':
+            expected_freqs = self.phonopy_expected_freqs["realsp_asr"]
+        elif "asr" in kwargs and kwargs["asr"] == "reciprocal":
+            expected_freqs = self.phonopy_expected_freqs["recip_asr"]
+        else:
+            expected_freqs = self.phonopy_expected_freqs["no_asr"]
+        qpts = np.array([
+            [0.00, 0.00, 0.00],
+            [0.00, 0.00, 0.50],
+            [-0.25, 0.50, 0.50],
+            [-0.151515, 0.575758, 0.5]
+        ])
+        qpoint_phonon_modes = fc.calculate_qpoint_phonon_modes(qpts, **kwargs)
+        npt.assert_allclose(
+            qpoint_phonon_modes.frequencies.to('hartree').magnitude,
+            expected_freqs.to('hartree').magnitude,
+            atol=1e-8
+        )
```

Comment:
Can we merge this with `test_force_constants_calculate_qpt_phonon_modes_castep.py`? Once files are read into a ForceConstants object, it shouldn't matter whether the data came from Phonopy or CASTEP (they were originally separate as a hack to get some tests in quickly, I can't say it was a good example...)

Reply:
I have consolidated these into the same file, written pytest fixtures that create the ForceConstants objects and then written one test function that loads all these fixture and checks their attributes